client/src/overview/components/QuantitySelector.jsx

```diff
@@ -1,22 +1,52 @@
 import React, {useEffect, useState} from 'react';
 
-let QuantitySelector = ({size, quantity}) => {
+let QuantitySelector = ({size, quantity, setPurchase}) => {
 
   const [amount, setAmount] = useState(1);
 
-  quantity = quantity > 15 ? 15 : quantity;
+  let stock = quantity > 15 ? 15 : quantity;
 
   let changeAmount = (e) => {
-    setAmount(prevState => e.target.value)
+    setAmount(prevState => {
+      setPurchase((prevState) => {
+        return {...prevState,
+          size: size,
+          quantity: e.target.value}
+      })
+      return e.target.value})
   }
 
+  useEffect(() => {
+    if (amount > stock) {
```

Comment:
@JeromeMTR what is the purpose of this conditional check? The way you have it written, it doesn't seem possible to set the amount higher than stock.

Reply:
@crypto-bender fixed resetting every time new size gets chosen

Reply:
@crypto-bender nevermind still having issues